src/daft-sql/src/planner.rs

```diff
@@ -913,7 +915,48 @@ impl SQLPlanner {
             SQLExpr::Map(_) => unsupported_sql_err!("MAP"),
             SQLExpr::Subscript { expr, subscript } => self.plan_subscript(expr, subscript.as_ref()),
             SQLExpr::Array(_) => unsupported_sql_err!("ARRAY"),
-            SQLExpr::Interval(_) => unsupported_sql_err!("INTERVAL"),
+            SQLExpr::Interval(interval) => {
+                let expr = self.plan_expr(&interval.value)?;
+                let time_unit = interval.leading_field.as_ref().ok_or_else(|| {
+                    PlannerError::invalid_operation("Interval must have a leading field")
+                })?;
+
+                let expr = expr
+                    .as_literal()
+                    .and_then(|lit| lit.as_str())
+                    .ok_or_else(|| {
+                        PlannerError::invalid_operation("Interval value must be a string")
+                    })?;
+
+                let count = expr
+                    .parse::<i64>()
+                    .map_err(|e| DaftError::ValueError(format!("Invalid interval count: {e}")))?;
```

Comment:
i think this'll likely fail for `interval '1 year'`. Maybe throw an `unsupported_sql_err` instead of a `ValueError` as we'll want to add support for this later on.